src/MemoryRouter.tsx

```diff
@@ -117,8 +115,22 @@ export class MemoryRouter extends BaseRouter {
       this.locale = options.locale;
     }
 
-    this.events.emit("routeChangeComplete");
+    this.events.emit("routeChangeStart", this.asPath, {
```

Comment:
According to the docs, it seems like the `routeChangeStart` should happen before the router is updated.  So maybe this `emit` should be moved higher in the file, before we update the fields?

Reply:
That could be right, I know it will send the route it's changing to as `url`, and yes within the same callback if I passed `asPath` from `useRouter` it was the old/previous path. Since changing to `routeChangeComplete`, I had to use the memoized `usePrevious` hook now so that makes sense.